frontend/src/concepts/modelRegistry/context/ModelRegistrySelectorContext.tsx

```diff
@@ -20,52 +20,66 @@ export const ModelRegistrySelectorContext = React.createContext<ModelRegistrySel
   updatePreferredModelRegistry: () => undefined,
 });
 
-export const ModelRegistrySelectorContextProvider =
-  conditionalArea<ModelRegistrySelectorContextProviderProps>(
-    SupportedArea.MODEL_REGISTRY,
-    true,
-  )(({ children }) => {
-    const [modelRegistries, isLoaded, error] = useModelRegistries();
-    const [preferredModelRegistry, setPreferredModelRegistry] =
-      React.useState<ModelRegistrySelectorContextType['preferredModelRegistry']>(undefined);
-
-    const firstModelRegistry = modelRegistries.length > 0 ? modelRegistries[0] : null;
+export const ModelRegistrySelectorContextProvider: React.FC<
+  ModelRegistrySelectorContextProviderProps
+> = ({ children, ...props }) => {
+  if (useModelRegistryEnabled()) {
+    return (
+      <EnabledModelRegistrySelectorContextProvider {...props}>
+        {children}
+      </EnabledModelRegistrySelectorContextProvider>
+    );
+  }
+  return children;
+};
 
-    React.useEffect(() => {
-      if (firstModelRegistry && !preferredModelRegistry) {
-        setPreferredModelRegistry(firstModelRegistry);
-      }
-    }, [firstModelRegistry, preferredModelRegistry]);
+export const EnabledModelRegistrySelectorContextProvider: React.FC<
+  ModelRegistrySelectorContextProviderProps
+> = ({ children }) => {
+  const [modelRegistries, isLoaded, error] = useModelRegistries();
+  const [preferredModelRegistry, setPreferredModelRegistry] =
+    React.useState<ModelRegistrySelectorContextType['preferredModelRegistry']>(undefined);
 
-    const updatePreferredModelRegistry = React.useCallback<
-      ModelRegistrySelectorContextType['updatePreferredModelRegistry']
-    >((modelRegistry) => {
-      setPreferredModelRegistry(modelRegistry);
-    }, []);
+  const firstModelRegistry = modelRegistries.length > 0 ? modelRegistries[0] : null;
 
-    if (!isLoaded) {
-      return <Bullseye>Loading model registries...</Bullseye>;
+  React.useEffect(() => {
+    if (firstModelRegistry && !preferredModelRegistry) {
+      setPreferredModelRegistry(firstModelRegistry);
     }
+  }, [firstModelRegistry, preferredModelRegistry]);
 
-    if (error) {
-      return (
-        <Bullseye>
-          <Alert title="Model registry load error" variant="danger" isInline>
-            {error.message}
-          </Alert>
-        </Bullseye>
-      );
-    }
+  const updatePreferredModelRegistry = React.useCallback<
+    ModelRegistrySelectorContextType['updatePreferredModelRegistry']
+  >((modelRegistry) => {
+    setPreferredModelRegistry(modelRegistry);
+  }, []);
 
+  const contextValue: ModelRegistrySelectorContextType = React.useMemo(
+    () => ({
+      modelRegistries,
+      preferredModelRegistry,
+      updatePreferredModelRegistry,
+    }),
+    [modelRegistries, preferredModelRegistry, updatePreferredModelRegistry],
+  );
+
+  if (!isLoaded) {
+    return <Bullseye>Loading model registries...</Bullseye>;
+  }
```

Comment:
With the context pulled up to to the top of the app, this is going to hold up the entire application waiting to load model registries where it isn't required.

Is there a more graceful way for dependant components to handle this?

Reply:
Ah, fair point. Will look into that.